pkg/api/validate/dynamic/encryptionathost.go

```diff
@@ -46,3 +53,20 @@ func (dv *dynamic) validateEncryptionAtHostSupport(VMSize api.VMSize, path strin
 
 	return nil
 }
+
+func (dv *dynamic) IsRegisteredForFeature(sub *api.SubscriptionProperties) error {
```

Comment:
minor nit:

The function name here doesn't make it clear _which_ feature we are checking for registration for. Given that it's a method on the generic dynamic validator, I would expect it to either be named specifically for the EncryptionAtHost feature (e.g. `IsSubRegisteredForEncryptionAtHostFeature`), or take in the feature name as a parameter if there are places to reuse it.